pandas/core/array_algos/masked_accumulations.py

```diff
@@ -0,0 +1,85 @@
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+
+from pandas.core.dtypes.common import (
+    is_bool_dtype,
+    is_float_dtype,
+    is_integer_dtype,
+)
+
+"""
+masked_accumulations.py is for accumulation algorithms using a mask-based approach
+for missing values.
+"""
+
+
+def _cum_func(
+    func: Callable,
+    values: np.ndarray,
+    mask: np.ndarray,
+    *,
+    skipna: bool = True,
+):
+    """
+    Accumulations for 1D masked array.
+
+    Parameters
+    ----------
+    func : np.cumsum, np.cumprod, np.maximum.accumulate, np.minimum.accumulate
+    values : np.ndarray
+        Numpy array with the values (can be of any dtype that support the
+        operation).
+    mask : np.ndarray
+        Boolean numpy array (True values indicate missing values).
+    skipna : bool, default True
+        Whether to skip NA.
+    """
+    dtype_info: np.iinfo | np.finfo
+    if is_float_dtype(values):
+        dtype_info = np.finfo(values.dtype.type)
+    elif is_integer_dtype(values):
+        dtype_info = np.iinfo(values.dtype.type)
+    elif is_bool_dtype(values):
+        dtype_info = np.iinfo(np.uint8)
```

Comment:
`np.iinfo(np.uint8).max` is 255. Wouldn't we want the max (fill value) to be 1?

Reply:
This does not matter, we are writing the fill value in a numpy array from dtype bool. The only relevant part is, that the minimum value is 0 and the maximum value is greater than zero

Reply:
Cool could you add this explanation as a comment here?

Reply:
Done

Reply:
Ah specifically that while in theory the max bool value is 1, the max uint8 is 255 and that's okay too

Reply:
Clarified the comment